### PR TITLE
Sprint 7 Staging

### PR DESCRIPTION
**Pull request by tamu-sad-iii**

### CODE REVIEW

File: src/app/core/service/rest.service.ts [hunk 1]
```diff
@@ -1,19 +1,33 @@
...
+
+export const REST_CACHE_TRANSLATE_STATE = makeStateKey<any>('REST_CACHE_TRANSLATE_STATE');
+
 @Injectable({
   providedIn: 'root',
 })
 export class RestService {
 
-  private cache: Map<string, any>;
+  private cache: any;
```
Comment (tamu-sad-iii) [inline src/app/core/service/rest.service.ts]: I am not sure this is a good idea or not. This in-memory cache will grow per server node process and each initial page request will transfer the entire cache to rehydrate in the browser. Best case, each rehydrated client performs the same UI interactions to make use of the cache. Worst case, each rehydrated client never utilizes the cache.

Comment (tamu-sad-iii) [reply src/app/core/service/rest.service.ts]: Selectively caching for state transfer would be ideal. Caching only async requests that are reused by all page requests and those necessary to handle unauthorized redirects.

Comment (tamu-sad-iii) [reply src/app/core/service/rest.service.ts]: Currently, every deep link to a different view request will cache async GET request responses on server and be rehydrated by each client browser.

Comment (tamu-sad-iii) [reply src/app/core/service/rest.service.ts]: To make for optimal demoing. Preload all pages intended to demo with a page refresh. Then next client rehydrate will have fast navigation through those pages.

**Review by Dbreck-TAMU — APPROVED** [review]

**Review by kaladay — APPROVED** [review]

**Pull request merged**